src/compiler/checker.ts

```diff
@@ -13213,6 +13213,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
     function isConstTypeVariable(type: Type): boolean {
         return !!(type.flags & TypeFlags.TypeParameter && some((type as TypeParameter).symbol?.declarations, d => hasSyntacticModifier(d, ModifierFlags.Const)) ||
+            isGenericTupleType(type) && findIndex(getTypeArguments(type), (t, i) => !!(type.target.elementFlags[i] & ElementFlags.Variadic) && isConstTypeVariable(t)) >= 0 ||

```

Comment:
Can you use something like `find`, `some`, or `forEach` instead of `findIndex`?

Reply:
`some` doesn't have the `i` param but `find` seems like it should work without the >= 0 check.

Reply:
In that case I'm probably fine with `findIndex` (or `findLastIndex` since variadics usually come at the end anyhow).